### PR TITLE
Removing the throw specifier from ArrayStoreAll constructor.

### PR DESCRIPTION
**Pull request by timothy-king**
Addresses CIDS: 1457252 and 1379620.

Miscellaneous cleanup to ArrayStoreAll.

### CODE REVIEW

File: src/expr/array_store_all.h [hunk 1]
```diff
@@ -18,64 +18,58 @@
...
 
 /**
  * Hash function for the ArrayStoreAll constants.
  */
 struct CVC4_PUBLIC ArrayStoreAllHashFunction {
   size_t operator()(const ArrayStoreAll& asa) const;
-};/* struct ArrayStoreAllHashFunction */
+}; /* struct ArrayStoreAllHashFunction */
+
+}  // namespace CVC4
```
Comment (aniemetz) [inline src/expr/array_store_all.h]: Ditto

File: src/expr/array_store_all.h [hunk 2]
```diff
@@ -18,64 +18,58 @@
...
-  // CONSTANT-kinded expression), and ArrayStoreAll needs Expr.
-  class Expr;
-  class ArrayType;
-}/* CVC4 namespace */
-
+// messy; Expr needs ArrayStoreAll (because it's the payload of a
+// CONSTANT-kinded expression), and ArrayStoreAll needs Expr.
+class Expr;
+class ArrayType;
+}  // namespace CVC4
```
Comment (aniemetz) [inline src/expr/array_store_all.h]: Ditto

File: src/expr/array_store_all.cpp [hunk 3]
```diff
@@ -96,4 +104,4 @@ size_t ArrayStoreAllHashFunction::operator()(const ArrayStoreAll& asa) const {
   return TypeHashFunction()(asa.getType()) * ExprHashFunction()(asa.getExpr());
 }
 
-} /* CVC4 namespace */
+}  // namespace CVC4
```
Comment (aniemetz) [inline src/expr/array_store_all.cpp]: We should be consistent here in terms of comment style, use /* */. We should change that in the clang format file.

Comment (timothy-king) [reply src/expr/array_store_all.cpp]: SG. I'm happy to use that as long as we can automate it.

**Comment by timothy-king** [comment]
@barrettcw This is a public header so updating these is not quite a no-op. (There was a related conversation in the slack channel.) I considered updating those functions to not be in scope for this PR.

**Comment by barrettcw** [comment]
@timothy-king OK sounds good.

**Pull request merged**